nodedb.go

```diff
@@ -606,7 +638,7 @@ func (ndb *nodeDB) getRoot(version int64) ([]byte, error) {
 			return nil, err
 		}
 		// TODO: maybe I shouldn't check in snapshot if it isn't here
-		if memroot != nil {
+		if len(memroot) != 0 {
```

Comment:
This change broke tests, reverting it.

```
    mutable_tree_test.go:145: 
        	Error Trace:	mutable_tree_test.go:145
        	Error:      	Expected nil, but got: version does not exist
        	Test:       	TestEmptyRecents
```

Reply:
Cool, but note the tests should be fixed, no? It's better to check length opposed to nil.

Reply:
But `nil` and `[]byte{}` are not the same thing. `nil` means the node does not exist, `[]byte{}` means the node exists but is empty. For root nodes, this is semantically different: a missing version vs an empty version.

Reply:
And `len(nil) == len([]byte{}) == 0`

Reply:
Right, I was under the impression we'd never store `nil`, hence we should check the length.

Reply:
Well, no, but this check is for whether the version exists in the recentDB, i.e. that it's not `nil`. The difference between `nil` and `[]byte{}` matters here.